source/common/stats/stats_impl.cc

```diff
@@ -57,6 +61,55 @@ std::string Utility::sanitizeStatsName(const std::string& name) {
   return stats_name;
 }
 
+TagExtractorImpl::TagExtractorImpl(const std::string& name, const std::string& regex)
+    : name_(name), regex_(regex) {}
+
+TagExtractorPtr TagExtractorImpl::createTagExtractor(const std::string& name,
+                                                     const std::string& regex) {
+
+  if (name.empty()) {
+    throw EnvoyException("tag_name cannot be empty");
+  }
+
+  if (!regex.empty()) {
+    return TagExtractorPtr{new TagExtractorImpl(name, regex)};
+  } else {
+    // Look up the default for that name.
+    const auto& name_regex_pairs = Config::TagNames::get().name_regex_pairs_;
```

Comment:
Could store this as a map to simplify (and make more scalable) this lookup..

Reply:
It was originally a map, and it was changed because we also wanted (somewhere else) to define the ordering for the default regexes.  To simplify the number of data structures and provide a single source of truth for both uses, I moved it to a vector of pairs.  This shouldn't be too problematic as these are only created on startup, and only needs to happen, at most, once per default regex, but if you think performance is an issue, we can add a more complex data structure or set of data structures.  We could also move the lookup into the `TagNames` class if we want to remove the burden from the caller (but right now this only needs to happen here).

Reply:
This is fine, no worries.